pages/documentation/how-pyth-works/cross-chain.mdx

```diff
@@ -1,12 +1,26 @@
 # Cross-chain
 
-Pyth uses a cross-chain mechanism to transfer prices from [Pythnet](pythnet.md) to target chains.
-The diagram below shows how prices are delivered from Pythnet to target chains:
+Pyth uses a cross-chain mechanism to transfer prices from [Pythnet](pythnet.md) to target chains. The diagram below
+shows how prices are delivered from Pythnet to target chains:
 
-![](../../../images/pythnet_schema.png)
+![](../../../images/Pull-Architecture-Hermes.png)
 
-Data providers publish their prices on Pythnet. The on-chain [oracle program](oracle-program.md) then aggregates prices for a feed to obtain the aggregate price and confidence. Next, the attester program regularly attests to the most recently observed Pyth prices and creates a Wormhole message to be sent to the Wormhole contract on Pythnet. The Wormhole guardians then observe the attestation message and create a signed VAA for the message.
+Data providers publish their prices on Pythnet. The on-chain [oracle program](oracle-program.md) then aggregates prices
+for a feed to obtain the aggregate price and confidence. Next, the Pythnet validators send a Wormhole message on each
+Pythnet slot to the Wormhole contract on Pythnet containing the Merkle root of all the prices. Wormhole guardians
+then observe the Merkle Root message and create a signed VAA for the message.
 
-The price service API continually listens to Wormhole for Pyth price update messages. It stores the latest update message in memory and exposes HTTP and websocket APIs for retrieving the latest update. (Anyone can run an instance of this webservice, but the Pyth Data Association runs a public instance for convenience.) When a user wants to use a Pyth price in a transaction, they retrieve the latest update message (a signed VAA) from the price service and submit it in their transaction. The target chain Pyth contract will verify the validity of the price update message and, if it is valid, store the new price in its on-chain storage. See [pull updates](../pythnet-price-feeds/on-demand.md) for more details about this process.
+[Hermes](../pythnet-price-feeds/hermes) continually listens to Wormhole for Pyth Merkle roots at each slot. It also
+listens to Pythnet to get all the price messages. It stores the latest price messages with their Merkle proof and Merkle
+root VAA in memory and exposes HTTP and websocket APIs for retrieving the latest update. (Anyone can run an instance of
+this webservice, but the Pyth Data Association runs a public instance for convenience.) When a user wants to use a Pyth
+price in a transaction, they retrieve the latest update message (prices with their Merkle proofs and Merkle root VAA)
```

Comment:
I think you probably want to explain how the update message works and why it's provably valid somewhere in this paragraph.